src/Database/Adapter/Memory.php

```
@@ -0,0 +1,1583 @@
+        $new = $this->filter($new);
+
+        if (!isset($this->data[$key]['indexes'][$old])) {
+            return true;
+        }
+
+        $this->data[$key]['indexes'][$new] = $this->data[$key]['indexes'][$old];
+        unset($this->data[$key]['indexes'][$old]);
+        return true;
+    }
+
+    public function createIndex(string $collection, string $id, string $type, array $attributes, array $lengths, array $orders, array $indexAttributeTypes = [], array $collation = [], int $ttl = 1): bool
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$indexAttributeTypes'. (undefined)

(UnusedFormalParameter)